client/my-sites/checkout/checkout/emergent-paywall-box.jsx

```diff
@@ -0,0 +1,246 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import page from 'page';
+import classNames from 'classnames';
+import { get, isEqual, debounce, startsWith, isBoolean } from 'lodash';
+import { connect } from 'react-redux';
+import { localize, translate } from 'i18n-calypso';
+import debug from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
```

Comment:
we never want to be importing `lib/analytics` outside of the singular analytics middleware. instead, we want to be dispatching actions which call for analytics either directly or preferably as metadata attached to some more semantically meaningful Redux dispatch…

would you mind circling around to update this code and remove the direct calls to analytics?
https://wpcalypso.wordpress.com/devdocs/client/state/analytics/README.md

Reply:
That's also great to have stated, cheers. I'll add this to the update too. 

If I get time I might look at other places that import `lib/analytics` directly for some easy janitorial victories - it looks like it happens in many places.

Given such frequency, and the suggestion that we should be 'never' do it this way, maybe we could literally do as you suggested and create deprecation/info alerts in the console (?)

Reply:
> That's also great to have stated

Sadly we completed this activity in [April, 2016](https://calypsop2.wordpress.com/2016/04/14/analytics-redux-middleware-redux/) but people really like `lib/analytics` and periodically bring it back.

It would do the project as a whole a service if you took on some more of these!